components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1790,22 +1819,13 @@ public void deleteGroup(String groupId) throws NotFoundException, CharonExceptio
                                 UserCoreUtil.isEveryoneRole(roleName, carbonUM.getRealmConfiguration())) {
                             continue;
                         }
-                        /**construct the group name with domain -if not already provided, in order to support
-                         multiple user store feature with SCIM.**/
-                        String groupNameWithDomain = null;
-                        if (roleName.indexOf(CarbonConstants.DOMAIN_SEPARATOR) > 0) {
-                            groupNameWithDomain = roleName;
-                        } else {
-                            groupNameWithDomain = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME + CarbonConstants.DOMAIN_SEPARATOR
-                                    + roleName;
-                        }
                         String userStoreDomainName = IdentityUtil.extractDomainFromName(roleName);
                         if (isInternalOrApplicationGroup(userStoreDomainName) || isSCIMEnabled(userStoreDomainName)) {
                             if (log.isDebugEnabled()) {
                                 log.debug("SCIM is enabled for the user-store domain : " + userStoreDomainName + ". " +
                                         "Including group with name : " + roleName + " in the response.");
                             }
-                            Group group = getGroupWithName(groupNameWithDomain);
+                            Group group = getGroupWithName(roleName);
```

Comment:
From where does the `roleName` is prepended with the domain?

Reply:
If the role not belongs to PRIMARY userstore, PRIMARY is not in the roleName. If we enable the properties in the identity.xml, Based on the properties PRIMARY will be prepended in the `getGroupWithName` method.